### PR TITLE
Build elm-plugin using gradle

### PR DESCRIPTION
**Pull request by mczubak**
This functionality enable of building elm-plugin using gradle. Please read carefully README. Because of limitation of grammar-kit generation of Elm.bnf parser can not be done on gradle site. That means generated parser needs to be stored directly in repo as it recommended by creators of grammar-kit. Most of created Intellij plug-ins, e.g haskforce, rust and go have parser classes stored in repo.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,13 +15,28 @@ This plugin is still at development phase. More features are coming soon.
 
 0. Setup your development environment according to [this instruction](http://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/setting_up_environment.html).
 0. Clone this repository.
-0. Open it as a Plugin Project in IntelliJ IDEA (either Community or Ultimate version).
+0. Type `gradle cleanIdea` in command line. All necessary dependencies will be installed.
```
Comment (durkiewicz) [inline README.md]: Maybe let's add a step:
```
0. Install Gradle (e.g. by `apt-get install gradle` on Ubuntu)
```

**Comment by durkiewicz** [comment]
I tried following the instruction in the README and I'm stuck on `gradle cleanIdea` step. I'm getting the following exception:
```
FAILURE: Build aborted because of an internal error.

* What went wrong:
Build aborted because of an unexpected internal error. Please file an issue at: http://forums.gradle.org.

* Try:
Run with --debug option to get additional debug info.

* Exception is:
org.gradle.api.internal.BeanDynamicObject$MetaClassAdapter$1: Cannot set the value of read-only property 'name' on root project 'elm-plugin'.
	at org.gradle.api.internal.BeanDynamicObject$MetaClassAdapter.setProperty(BeanDynamicObject.java:169)
	at org.gradle.api.internal.BeanDynamicObject.setProperty(BeanDynamicObject.java:107)
	at org.gradle.api.internal.CompositeDynamicObject.setProperty(CompositeDynamicObject.java:101)
	at org.gradle.api.internal.project.DefaultProject_Decorated.setProperty(Unknown Source)
	at org.gradle.initialization.ProjectPropertySettingBuildLoader.addPropertiesToProject(ProjectPropertySettingBuildLoader.java:75)
	at org.gradle.initialization.ProjectPropertySettingBuildLoader.setProjectProperties(ProjectPropertySettingBuildLoader.java:53)
	at org.gradle.initialization.ProjectPropertySettingBuildLoader.load(ProjectPropertySettingBuildLoader.java:49)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:138)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:113)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:81)
	at org.gradle.launcher.cli.ExecuteBuildAction.run(ExecuteBuildAction.java:38)
	at org.gradle.launcher.exec.InProcessGradleLauncherActionExecuter.execute(InProcessGradleLauncherActionExecuter.java:39)
	at org.gradle.launcher.exec.InProcessGradleLauncherActionExecuter.execute(InProcessGradleLauncherActionExecuter.java:25)
	at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:50)
	at org.gradle.api.internal.Actions$RunnableActionAdapter.execute(Actions.java:171)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:201)
	at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:174)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:170)
	at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:139)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
	at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
	at org.gradle.launcher.Main.doAction(Main.java:48)
	at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
	at org.gradle.launcher.Main.main(Main.java:39)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:50)
	at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:32)
	at org.gradle.launcher.GradleMain.main(GradleMain.java:26)


BUILD FAILED

Total time: 2.148 secs
```

**Comment by ianp** [comment]
Given that we apparently can't run Grammar Kit from Gradle (due to JetBrains/Grammar-Kit#35) does it make sense to have those tasks in the build file? Maybe just (as you suggest) check the `gen` folder into Git and update the readme accordingly?

**Comment by durkiewicz** [comment]
Thank you @mczubak for your work. I'm sorry that it's not going to be used at the moment but due to the limitations of the IntelliJ plugin development ecosystem's limitations I'm not going to integrate it. In my opinion, there are too many cons / negative implications.

**Comment by paplorinc** [comment]
@durkiewicz, @mczubak, the rust plugin for Idea generates it by copying the .jar file in the project and running it as a java task: https://github.com/JetBrains/gradle-intellij-plugin/issues/34#issuecomment-220818975

**Comment by bgourlie** [comment]
FWIW, the Intellij-Rust plugin represents what I'd consider idiomatic practices as it's maintained and actively developed by an actual JetBrains employee.

**Pull request closed without merging**